### PR TITLE
Add option to add custom microservices

### LINKED ISSUE #1169: Add option to add custom microservices
**Is your feature request related to a problem? Please describe.**
The United Manufacturing Hub provides an IT / OT infrastructure, therefore it should be able to provide custom microservices. Kuebrnetes is quite complex and might hinder someone in deploying microservices and then in total in using the United Manufacturing Hub.

**Describe the solution you'd like**
Add functionality to Helm Chart (allowing multiple microservices to be deployed):
```
customMicroservice:
                - hello123:
                                image: united-manufacturing-hub/custom-service:latest
                                
                                # imagePullPolicy: Always is recommended when using latest tags. Otherwise, please use IfNotPresent
                                imagePullPolicy: Always
                                
                                # these will be passed as environment variables
                                env:
                                                - name: LOGGING_LEVEL
                                                value: PRODUCTION
                                
                                # ports will expose port 80 on the host on port 8080
                                ports: "8080:80"
                                
                                # startupProbe, livenessProbe, check whether port 8080 "/health" is accessible. 
                                # If not (e.g., due to container stuck), the container will be restarted
                                # startupProbe: 100
                                # livenessProbe: 50
                                
                                # statefulEnabled will automatically mount a volume under /data, which can be used to store data. 
                                # Recommendation is to turn it off
                                statefulEnabled: true
                - test123:
                                image: ……
                                …
```

### PR DESCRIPTION
**Pull request by Scarjit**
# Description

This pr adds an option to easily deploy custom microservices inside the UMH stack

Fixes #1169 

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [X] AutoDeploy

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [X] I have checked that my issue complies with the [Contributing guidelines](https://github.com/united-manufacturing-hub/united-manufacturing-hub/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Comment by Scarjit** [comment]
Leaving this as draft for more testing

File: deployment/united-manufacturing-hub/values.yaml [hunk 1]
```diff
@@ -961,6 +961,55 @@ _000_commonConfig:
...
+      # these will be passed as environment variables
+      env:
+        - name: LOGGING_LEVEL
+          value: PRODUCTION
+
+      # this will expose port 80 on the host on port 8080
+      port: 81
+      externalPort: 8081
+
+      # startupProbe, livenessProbe, check whether port 8080 "/health" is accessible.
```
Comment (JeremyTheocharis) [inline deployment/united-manufacturing-hub/values.yaml:1005]: In the current implementation, it checks for the variable 'port'. Lets set that to port 8080 or probePort

**Review by JeremyTheocharis — CHANGES_REQUESTED** [review]

File: deployment/united-manufacturing-hub/values.yaml [hunk 2]
```diff
@@ -961,6 +961,55 @@ _000_commonConfig:
...
+      externalPort: 8080
+
+      # startupProbe, livenessProbe, check whether port 8080 "/health" is accessible.
+      # If not (e.g., due to container stuck), the container will be restarted
+      # startupProbe: 100
+      # livenessProbe: 50
+
+      # statefulEnabled will automatically mount a volume under /data, which can be used to store data.
+      # Recommendation is to turn it off
+      statefulEnabled: true
```
[the remark this answers is not in the document]
Comment (Scarjit) [reply deployment/united-manufacturing-hub/values.yaml:988]: Not yet

[the remark this answers is not in the document]
Comment (Scarjit) [reply deployment/united-manufacturing-hub/values.yaml:989]: Resolved in 49ff0dc53a18051400e66102ca7e9ee6e0647e56

File: deployment/united-manufacturing-hub/values.yaml (hunk 1 above)
Comment (Scarjit) [reply deployment/united-manufacturing-hub/values.yaml]: 49ff0dc53a18051400e66102ca7e9ee6e0647e56

**Review by JeremyTheocharis — APPROVED** [review]

**Pull request merged**